update-reviewers.py

```diff
@@ -0,0 +1,200 @@
+"""
+This script parses through our reviews and contributors and
+
+1. Updates reviewer, editor and maintainer data in the contributor.yml file to
+ensure all packages they supported are listed there.
+1b: And that they have a listing as peer-review under contributor type
+2. Updates the packages metadata with the participants names if it's missing
+3. FUTURE: finally it looks to see if we are missing review participants from
+the review issues in the contributor file and updates that file.
+"""
+
+# TODO: right now this just updates all packages. but in some cases, we may
+# update the file itself (how often??) so it's worth considering a way to
+# just update the current file and flag entries that we do not want to update
+
+import pickle
+
+from pyosmeta.contributors import ProcessContributors
+from pyosmeta.file_io import get_api_token
+
+API_TOKEN = get_api_token()
+process_contribs = ProcessContributors([], [], API_TOKEN)
+# Open the contributor output dict stored in a pickle after the
+# parse_contributors script is run. key  = gh_username lowercase
+with open("all_contribs.pickle", "rb") as f:
+    contribs = pickle.load(f)
+
+# Open packages yaml created by running parse_review_issues.py
+with open("all_reviews.pickle", "rb") as f:
+    packages = pickle.load(f)
+
+# TODO: add a flag for the person who was submitting in contributor type??
+
+# TODO: It isn't updating the name of the gh user in the packages file.
+# def _find_name(username: str, contribs: dict):
+#     """in progress not used yetA helper to find..."""
+#     if aPackage[aType]["name"] == "":
+#         try:
+#             # Try to populate name from contributors file
+#             aPackage[aType]["name"] = contribs[gh_user]["name"]
+#         except KeyError as ke:
+#             #  If there is a keyerror - add them to the contribs dict
+#             missing_contribs.append(gh_user)
+
+
+# TODO: Add maintainer, reviewer, editor to contributor type.
+ppl_types = {
+    # TODO: make sure we can add a 3rd or 4th reviewer - crowsetta has this as
+    # will biocypher
+    "reviewer_1": ["packages-reviewed", "reviewer"],
+    "reviewer_2": ["packages-reviewed", "reviewer"],
+    "editor": ["packages-editor", "editor"],
+    "submitting_author": [
+        "packages-submitted",
+        ["maintainer", "submitting-author"],
+    ],
+}
+
+
+missing_contribs = []
+for pkgName, issue_meta in packages.items():
+    print(pkgName)
+    # Get review meta -
+    # start with reviewers and editor
+    for issue_review_role in ppl_types:
```

Comment:
This section has many conditionals because sometimes the field is empty, other times it has existing values and i need to check whether the value exists before adding it so there aren't duplicates. FINALLY sometimes the user doesn't exist so i need to add them if that is the case